src/temgymbasic/components.py

```diff
@@ -418,6 +462,46 @@ def get_image(self, rays: Rays) -> NDArray:
         )
         return image
 
+    def get_image_intensity(self, rays: Rays) -> NDArray:
```

Comment:
Seems like there is a fair amount of duplication with `get_image`, we can refactor the common parts.